torch/autograd/profiler.py

```diff
@@ -101,55 +101,39 @@ def export_chrome_trace(self, path):
         Arguments:
             path (str): Path where the trace will be written.
         """
-        import json
+        import os
         with open(path, 'w') as f:
             chrome_events = []
             next_id = 0
+            # Use file IO over using json.dump since JSON dumping is very slow and
+            # this technique is proven to give a 4x speedup.
+            f.write("[")
             for evt in self:
-                chrome_events.append(dict(
-                    name=evt.name,
-                    ph='X',
-                    ts=evt.cpu_interval.start,
-                    dur=evt.cpu_interval.elapsed_us(),
-                    tid=evt.thread,
-                    pid='CPU functions',
-                    args={},
-                ))
+                f.write('{"name": "%s", "ph": "X", "ts": %s, ' \
+                        '"dur": %s, "tid": %s, "pid": "CPU functions", ' \
+                        '"args": {}}, ' % (evt.name, evt.cpu_interval.start,
+                                             evt.cpu_interval.elapsed_us(), evt.thread))
```

Comment:
I would suggest to write this in a more readable form, one parameter per line:
```
f.write('{"name": "%s", '
         '"ph": "s", '
         '"ts": %s, '
         '"tid": %s, '
         '"pid": "CPU functions", '
         '"id": %s, '
         '"cat": "cpu_to_cuda", '
         '"args": {}}, ' % (evt.name,
                            evt.cpu_interval.start,
                            evt.thread,
                            next_id))
```

Reply:
And same for the other `f.write`'s.

Note that you don't need `\` to concatenate strings, as long as they're in `( )` it works over multiple lines.